static/app/components/globalDrawer/index.spec.tsx

```diff
@@ -0,0 +1,207 @@
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+    expect(screen.getByRole('complementary', {name: ariaLabel})).toBeInTheDocument();
+
+    await userEvent.click(screen.getByTestId('drawer-test-close'));
+    await waitForDrawerToHide(ariaLabel);
+
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('complementary', {name: ariaLabel})
+    ).not.toBeInTheDocument();
+  });
+
+  it('calls onClose handler when close button is clicked', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose button</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+
+    await userEvent.click(screen.getByRole('button', {name: 'Close Drawer'}));
+    await waitForDrawerToHide();
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+  });
+
+  it('calls onClose handler when clicking outside the drawer', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose outside click</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+
+    await userEvent.click(screen.getByTestId('drawer-test-outside'));
+    await waitForDrawerToHide();
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+  });
+
+  it('calls onClose handler when escape key is pressed', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose escape</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    const content = screen.getByTestId('drawer-test-content');
+    expect(content).toBeInTheDocument();
+
+    // TODO(Leander): Implement the following test case when `useHotKeys` hook allows for userEvent signals
+    // await userEvent.keyboard('{Escape}');
+    // await waitForDrawerToHide();
+
+    // expect(closeSpy).toHaveBeenCalled();
+    // expect(content).not.toBeInTheDocument();
+  });
```

Comment:
I dont think this is testing what anything new, you already assert drawer-test-content presence in other tests. Feel free to mark the test as todo or remove it in this case. I think the only thing that isnt tested is the closing on location change, but that is trivial.